L/liboqs/build_tarballs.jl

```diff
@@ -0,0 +1,187 @@
+# override the CPU/ABI chosen by the toolchain and can select unsupported ISAs or break
+# assembly. The cross toolchain should be the sole source of ISA/ABI flags, so strip them.
+# (Together with -DOQS_OPT_TARGET=generic this forces portable code paths across targets.)
+LC_ALL=C sed -i.bak -E \
+  's/-march=[^" )]+//g; s/-mcpu=[^" )]+//g; s/-mtune=[^" )]+//g' \
+  .CMake/*.cmake cmake/*.cmake 2>/dev/null || true
+
+mkdir -p build && cd build
+
+# ---- Darwin flags (kept for completeness; no-op on Windows builds) ----
+APPLE_FLAGS=()
+if [[ "${target}" == *apple-darwin* ]]; then
+  if [[ "${target}" == aarch64-apple-darwin* ]]; then DARWIN_CPU=arm64; else DARWIN_CPU=x86_64; fi
+  APPLE_FLAGS+=(
+    -DCMAKE_SYSTEM_NAME=Darwin
+    -DCMAKE_SYSTEM_VERSION=20
+    -DCMAKE_OSX_DEPLOYMENT_TARGET=11.0
+    -DCMAKE_SHARED_LIBRARY_SUFFIX=.dylib
+    -DCMAKE_SYSTEM_PROCESSOR=${DARWIN_CPU}
+    -DCMAKE_OSX_ARCHITECTURES=${DARWIN_CPU}
+    -DCMAKE_INSTALL_NAME_DIR=@rpath
+    -DCMAKE_INSTALL_RPATH=@rpath
+    -DOQS_PERMIT_UNSUPPORTED_ARCHITECTURE=ON
+  )
+  export MACOSX_DEPLOYMENT_TARGET=11.0
+fi
+
+# ---- Use clang/lld on Linux/BSD/Windows targets ----
+EXTRA_FLAGS=()
+if [[ "${target}" == *-linux-* || "${target}" == *-freebsd* || "${target}" == *-w64-mingw32 ]]; then
+  SYSROOT="/opt/${target}/${target}/sys-root"
+  EXTRA_FLAGS+=(
+    -DCMAKE_C_COMPILER=clang
+    -DCMAKE_C_COMPILER_TARGET=${target}
+    -DCMAKE_SYSROOT=${SYSROOT}
+    -DCMAKE_AR=$(command -v llvm-ar)
+    -DCMAKE_RANLIB=$(command -v llvm-ranlib)
+    -DCMAKE_NM=$(command -v llvm-nm)
+  )
+
+  # ---- extra for MinGW ----
+  if [[ "${target}" == *-w64-mingw32 ]]; then
+    [[ -x "$(command -v llvm-rc)" ]] && EXTRA_FLAGS+=(-DCMAKE_RC_COMPILER=$(command -v llvm-rc))
+    EXTRA_FLAGS+=(
+      -DCMAKE_SYSTEM_NAME=Windows
+      -DCMAKE_SYSTEM_VERSION=10
+      -DOQS_PERMIT_UNSUPPORTED_ARCHITECTURE=ON
+    )
+  fi
+
+  if command -v ld.lld >/dev/null 2>&1; then
+    export LD=$(command -v ld.lld)
+    export LDFLAGS="-fuse-ld=lld ${LDFLAGS}"
+    EXTRA_FLAGS+=(
+      -DCMAKE_LINKER=${LD}
+      -DCMAKE_EXE_LINKER_FLAGS="${LDFLAGS}"
+      -DCMAKE_SHARED_LINKER_FLAGS="${LDFLAGS}"
+      -DCMAKE_MODULE_LINKER_FLAGS="${LDFLAGS}"
+    )
+  fi
+
+  export CC=clang
+  export AR=$(command -v llvm-ar)
+  export RANLIB=$(command -v llvm-ranlib)
+  export NM=$(command -v llvm-nm)
+  export CFLAGS="--target=${target} --sysroot=${SYSROOT} ${CFLAGS}"
+  export ASFLAGS="--target=${target} --sysroot=${SYSROOT} ${ASFLAGS}"
+  # Disable GOTPCRELX relaxation on x86/x86_64 to allow linking with older bfd ld
+  if [[ "${target}" == x86_64-* || "${target}" == i686-* ]]; then
+    export CFLAGS="${CFLAGS} -Wa,-mrelax-relocations=no"
+    export CXXFLAGS="${CXXFLAGS} -Wa,-mrelax-relocations=no"
+    # Also apply to plain assembly
+    export ASFLAGS="${ASFLAGS} -mrelax-relocations=no"
+  fi
+fi
+# ----------------------------
```

Comment:
Why do you need all of this?  Why not using `-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}`, as suggested in [the documentation](https://docs.binarybuilder.org/stable/build_tips/#CMake-builds), which would do the right thing?

Reply:
Oh, I missed the instruction manual. I'll try `-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}`. Thank you for your advice!!!